libraries/core-react/src/Table/Cell.tsx

```diff
@@ -42,19 +42,23 @@ const TableBase = styled.td`
   ${Base}
 `
 
-type Props = {
+type CellProps = {
   /** Specifies which td or th to use */
-  as: 'td' | 'th'
+  as?: 'td' | 'th'
   /** Specifies which variant to use */
-  variant: 'text' | 'icon' | 'numeric' | 'input'
-} & HTMLAttributes<HTMLTableCellElement>
```

Comment:
Just thought of one thing. How will removal of `HTMLAttributes<HTMLTableCellElement>` affect other `react` props as `style`, `className` or `forwardedAs`?

Reply:
Good point, I’ll add those manually.

Reply:
What was the reason for removing `HTMLAttributes<HTMLTableCellElement>`? There are a plethora of [other attributes ](https://developer.mozilla.org/en-US/docs/Web/API/HTMLTableCellElement) we miss out on by not having this.

Reply:
The reason was that the more I dug into this the more I realised the component needs a more thorough rewrite, so I focused on getting the story to work, since we’ll have to refactor the component anyway before we merge typescript into develop. It doesn’t make sense for a table cell to have the scope attribute for example.

Reply:
![image](https://user-images.githubusercontent.com/2081882/97555764-d14c4900-19d8-11eb-8bc2-897098118d15.png)

This is what I get if I use `HTMLAttributes<HTMLTableCellElement>`. So ideally the types should be based on the value of the `variant` prop, which I started looking into using function overload, but decided to save for the [refactoring issue](https://github.com/equinor/design-system/issues/297#issuecomment-717852095).

Reply:
I did some testing. Since we have the option to have a cell as `th` or `td`, what we are talking about is a union type. So then something along the lines of this makes more sense:

```typescript
type Props = {
  /** Specifies which td or th to use */
  as: 'td' | 'th'
  /** Specifies which variant to use */
  variant: 'text' | 'icon' | 'numeric' | 'input'
} & (
  | TdHTMLAttributes<HTMLTableDataCellElement>
  | ThHTMLAttributes<HTMLTableHeaderCellElement>
)
```

Reply:
That seems to do the trick 👍 But what’s the difference then between `HTMLAttributes<HTMLTableCellElement>` and `TdHTMLAttributes<HTMLTableDataCellElement>`? 🤔 I find this confusing…

Reply:
As far as I have understood the latter is more specific, it extends from [HTMLAttributes and adds the td specific attributes in addition.](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts)

Reply:
I see… and it has [all the same props as ThHTMLAttributes](ThHTMLAttributes), so then that will probably suffice.

Reply:
> I find this confusing…

Me to 🙈  

You can find out a lot by looking at the types. Looking at how [JSX.Intrinsict](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/5454930b2034878b7cf0d749a1272a5e37f41bfa/types/react/index.d.ts#L2985) can tell you a lot  

I think it was wrong of us to use `HTMLAttributes<>` on many of the components to begin with, as they don't actually don extend the specific attributes for the dom types we pass into them, the type is only applied to dom callback handlers. 

The more specific Attributes such as `TdHTMLAttributes` adds the correct attributes ontop of setting the correct dom callbacks.

Reply:
I see, we should add something about that in the discussions thread then, I can do that later.